experimental/packages/shim-opencensus/src/shim.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag, trace, Tracer } from '@opentelemetry/api';
+import { Hook } from 'require-in-the-middle';
+import * as oc from '@opencensus/core';
+
+import { ShimTracer } from './ShimTracer';
+import { VERSION } from './version';
+
+type CoreTracerConstructor = new (
+  ...args: ConstructorParameters<typeof oc.CoreTracer>
+) => oc.Tracer;
+
+let hook: Hook | null = null;
+
+interface OpenCensusShimConfig {
+  /**
+   * An optional OpenTelemetry tracer to send OpenCensus spans to. If not provided, one will be
+   * created for you.
+   */
+  tracer?: Tracer | undefined;
+}
+
+/**
+ * Patches OpenCensus to redirect all instrumentation to OpenTelemetry. Uses
+ * require-in-the-middle to override the implementation of OpenCensus's CoreTracer.
+ *
+ * Use {@link uninstallShim} to undo the effects of this function.
+ *
+ * @param config
+ */
+export function installShim({
+  tracer = trace.getTracer('@opentelemetry/shim-opencensus', VERSION),
+}: OpenCensusShimConfig = {}): void {
+  diag.info('Installing OpenCensus shim require-in-the-middle hook');
+
+  hook = new Hook(['@opencensus/core'], exports => {
```

Comment:
Discussed here with @Flarna https://github.com/open-telemetry/opentelemetry-js/pull/3778#discussion_r1185765040:

> as far as I remember one reason to have the `@opentelemetry/instrumentation` package is to avoid that several modules do patching on their own and depend on RITM (and later import in the middle to get ESM support). There is also some optimization/tuning there to use a single RITM hook for performance reasons.
> 
> Not sure if it is possible/useful but maybe this could be changed to use `@opentelemetry/instrumentation` instead RITM directly?

However, I ran into some issues with `@opentelemetry/instrumentation` discussed in that thread. Flarna's suggestion then was

> I guess the solution would be to patch at the source of `CoreTracer` - the exports of `@opencensus/core/build/src/trace/model/tracer.js`. This should be a normal export.
> 
> Patching this one has the benefit that all imports of `CoreTracer` are effected (not only the reexport in index.js) - in case there are others.

Patching worked, but unwrapping in the `Instrumentation.disable()` (mainly needed for test code) doesn't work correctly because of the require cache. I tried clearing it but still ran into issues I couldn't figure out. How important is this?

Reply:
The unpatch is generally for testing and yes we regularly discuss removing it. The best way to disable otel is with a sampler (or other signal-specific component). Unpatch needs to work well enough to unpatch the module for testing, but already patched objects that users may have references to are typically dropped at the conclusion of each test.

Reply:
Ok lets discuss in the SIG. I'm not sure the best way to move forward here.

Reply:
Summary from the SIG

- Too many RITM hooks were expensive because the hooks get chained for every `require()` call.
- Having this one additional hook is OK for now, we can revisit if needed.

Reply:
Fine for me to keep RITM here.
But as said RITM doesn't support ESM modules therefore this limitations should be mentioned in the docs.

Reply:
@Flarna are you OK with this PR as is given that OpenCensus doesn't support ESM? It should be a moot point

Reply:
ah yes, fine for me. I was not aware that OpenCensus can't be used with ESM.